app/src/main/java/com/example/myapplication/ui/MainViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.myapplication.ui
+
+import androidx.lifecycle.ViewModel
+import com.example.myapplication.model.ApiHandler
+import com.example.myapplication.model.Product
+import com.example.myapplication.model.SimilarProductsInfo
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class MainViewModel : ViewModel() {
+    private val apiHandler = ApiHandler()
+
+    private val _productState = MutableStateFlow(Product())
+    val productState: StateFlow<Product> get() = _productState
```

Comment:
I think this is ok for now, but if you want you can try out something :) when we want to store a state of a UI we mostly create a data class for UI's state, something like that "MainUiState". On the view (in this case MainActivity) you can just collect changes from this flow. So what is it look like in practice -> MainUiState will hold similar products or the title of the screen or the product obj etc. it depends on the screen or how you separate the essential data. When you received the result of the product or similar products you just update MainUiState with them.